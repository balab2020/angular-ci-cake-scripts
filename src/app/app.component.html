<h1>Hello World</h1>
<app-heoow-jest></app-heoow-jest>
<div class="content">
    <span>angular-hello app is running!</span>
</div>