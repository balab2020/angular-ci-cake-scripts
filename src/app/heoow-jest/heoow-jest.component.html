<p>Testing Jest framework easy your life !!!!!!!!!</p>