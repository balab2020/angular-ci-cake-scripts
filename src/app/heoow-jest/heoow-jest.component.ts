import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-heoow-jest',
  templateUrl: './heoow-jest.component.html',
  styleUrls: ['./heoow-jest.component.scss']
})
export class HeoowJestComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

}
