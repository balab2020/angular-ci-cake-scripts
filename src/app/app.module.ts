import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { HeoowJestComponent } from './heoow-jest/heoow-jest.component';

@NgModule({
  declarations: [
    AppComponent,
    HeoowJestComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
