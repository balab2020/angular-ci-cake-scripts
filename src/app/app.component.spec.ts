import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { HeoowJestComponent } from './heoow-jest/heoow-jest.component';
describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent, HeoowJestComponent
      ],
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(fixture.nativeElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    //@ts-ignore
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it(`should have as title 'angular-hello'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('angular-hello');
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(compiled.querySelector('.content span').textContent).toContain('angular-hello app is running!');
  });
});
