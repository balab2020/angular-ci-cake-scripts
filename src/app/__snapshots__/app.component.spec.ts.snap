// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<span
  _ngcontent-a-c3=""
>
  angular-hello app is running!
</span>
`;
